cmd/agent/subcommands/run/command.go

```diff
@@ -633,6 +619,12 @@ func startAgent(
 	// start dependent services
 	go startDependentServices()
 
+	if err := otelcollector.Start(); err != nil {
+		return err
+	}
+	// TODO: remove this once migrating the status package to components
```

Comment:
Can you use `// TODO: (components)` to prefix the TODO comments related the component framework so we can list them easily?